sklearn/metrics/tests/test_common.py

```diff
@@ -799,30 +799,52 @@ def test_thresholded_invariance_string_vs_numbers_labels(name):
                 metric(y1_str.astype('O'), y2)
 
 
-invalids = [([0, 1], [np.inf, np.inf]),
-            ([0, 1], [np.nan, np.nan]),
-            ([0, 1], [np.nan, np.inf])]
-
+invalids_nan_inf = [
+    ([0, 1], [np.inf, np.inf]),
+    ([0, 1], [np.nan, np.nan]),
+    ([0, 1], [np.nan, np.inf]),
+    ([0, 1], [np.inf, 1]),
+    ([0, 1], [np.nan, 1]),
+]
 
 @pytest.mark.parametrize(
         'metric',
         chain(THRESHOLDED_METRICS.values(), REGRESSION_METRICS.values()))
 def test_regression_thresholded_inf_nan_input(metric):
-
-    for y_true, y_score in invalids:
+    for y_true, y_score in invalids_nan_inf:
         with pytest.raises(ValueError, match="contains NaN, infinity"):
             metric(y_true, y_score)
 
 
 @pytest.mark.parametrize('metric', CLASSIFICATION_METRICS.values())
 def test_classification_inf_nan_input(metric):
-    # Classification metrics all raise a mixed input exception
+    # check that classification metrics raise a message mentioning the
+    # occurrence of non-finite values in the target vectors.
```

Comment:
Slowly nudging to use docstrings for comments, so we can start enforcing comments for test?

```suggestion
    """check that classification metrics raise a message mentioning the
    occurrence of non-finite values in the target vectors."""
```